modules/common/src/main/java/org/opencastproject/mediapackage/AdaptivePlaylist.java

```diff
@@ -159,10 +162,12 @@ static boolean checkForMaster(File file) throws IOException {
   static boolean checkForVariant(File file) throws IOException {
     if (!isPlaylist(file))
       return false;
-    Optional<Matcher> m = Files.lines(file.toPath()).map(variantPatt::matcher).filter(Matcher::find).findFirst();
-    if (m.isPresent()) {
-      m = null;
-      return true;
+    try (Stream<String> lines = Files.lines(file.toPath())) {
+      Optional<Matcher> m = lines.map(variantPatt::matcher).filter(Matcher::find).findFirst();
+      if (m.isPresent()) {
+        m = null;
+        return true;
+      }
```

Comment:
Why set `m = null`? It's a local variable anyway and should be discarded due to its scope.
This also would allow fur the much simpler:

```suggestion
    try (Stream<String> lines = Files.lines(file.toPath())) {
      return lines
          .map(masterPatt::matcher)
          .filter(Matcher::find)
          .findFirst()
          .isPresent();
    }
```

I also changed the formatting since I think that's easier to read like this, but that might just be my preference.

Reply:
Thinking about this for a moment, isn't this simply:

```java
return lines
    .map(masterPatt::matcher)
    .anyMatch(Matcher::find);
```

Reply:
I suppose so. This code was there before and I did not change it, just fixed the bug. I can make the requested change.